Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -315,6 +344,215 @@ extension BoringSSLRSAPublicKey {
         }
 
+        fileprivate func blind<H: HashFunction>(
+            _ message: _RSA.BlindSigning.PreparedMessage,
+            parameters: _RSA.BlindSigning.Parameters<H>
+        ) throws -> (blindedMessage: Data, blindInverse: _RSA.BlindSigning.BlindInverse) {
+            /// ```
+            /// All BN_CTX_get() calls must be made before calling any other functions that use the ctx as an argument.
+            /// ...
+            /// BN_CTX_get() returns a pointer to the BIGNUM, or NULL on error. Once BN_CTX_get() has failed, the
+            /// subsequent calls will return NULL as well, so it is sufficient to check the return value of the last
+            /// BN_CTX_get() call.
+            /// ```
+            /// —— Extract from `man 3 BN_CTX_get`.
+            let bnCtx = CCryptoBoringSSL_BN_CTX_new()
+            CCryptoBoringSSL_BN_CTX_start(bnCtx)
+            defer {
+                CCryptoBoringSSL_BN_CTX_end(bnCtx)
+                CCryptoBoringSSL_BN_CTX_free(bnCtx)
+            }
+            let m = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let gcd = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let r = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let inv = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let x = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let z = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            guard z != nil else { throw CryptoKitError.internalBoringSSLError() }
+
+            let rsaPublicKey = CCryptoBoringSSL_EVP_PKEY_get0_RSA(self.pointer)
+            let outputSize = Int(CCryptoBoringSSL_RSA_size(rsaPublicKey))
+
+            // 1. encoded_msg = EMSA-PSS-ENCODE(msg, bit_len(n)) with Hash, MGF, and salt_len as defined in the parameters
+            // 2. If EMSA-PSS-ENCODE raises an error, re-raise the error and stop
+            let hashDigestType = try! DigestType(forDigestType: H.Digest.self)
+            let hash = H.hash(data: message.rawRepresentation)
+            let saltLength: Int32
+            switch parameters.padding {
+            case .PSS: saltLength = Int32(H.Digest.byteCount)
+            case .PSSZERO: saltLength = 0
+            }
+            var encodedMessage = [UInt8](repeating: 0, count: outputSize)
+            guard hash.withUnsafeBytes({ hashBufferPtr in
+                CCryptoBoringSSL_RSA_padding_add_PKCS1_PSS_mgf1(
+                    rsaPublicKey,
+                    &encodedMessage,
+                    hashBufferPtr.baseAddress,
+                    hashDigestType.dispatchTable,
+                    hashDigestType.dispatchTable,
+                    saltLength
+                )
+            }) == 1 else {
+                switch ERR_GET_REASON(CCryptoBoringSSL_ERR_get_error()) {
+                case RSA_R_DATA_TOO_LARGE_FOR_KEY_SIZE:
+                    throw _RSA.BlindSigning.Error.messageTooLong
+                default:
+                    throw CryptoKitError.internalBoringSSLError()
+                }
+            }
```

Comment:
Can I propose factoring this section (and probably several others)) out to small helper functions? In fact, you could create a little enum-based namespace and turn each of these steps into functions in that namespace. That may make it a touch easier to follow, as this method has gotten very long.

Reply:
OK, I've factored this and a bunch of other things into a `fileprivate enum BlindSigningHelpers`. Hopefully things should read a bit clearer now.

Reply:
I think we can still go further here. In particular, the unsafe temporary allocations can probably be factored into safer helper methods.

Reply:
OK, I tried to find the right balance of keeping the steps clear from the RFC and factoring out the unsafe bits so I think the flow of this is now sensible. Specifically, if I have separate helpers for the separate steps from the RFC commentary, EEMSA-PSS-ENCODE, and bytes_to_int, then I must have a temporary allocation here.

To help converge this though, I've pushed these into one helper, abstracting over steps 1-3 which does remove this one last use of with unsafe temporary allocation from this function.